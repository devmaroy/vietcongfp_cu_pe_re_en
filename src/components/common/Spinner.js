import React from 'react';
import PropTypes from 'prop-types';

const Spinner = ( { imagePath, alt, style, classNameWrapper, classNameImg } ) => (
    <div className={ classNameWrapper }>
        <img 
            style={ style }
            src={ imagePath } 
            alt={ alt }
            className={ classNameImg }
        />
    </div>
);

Spinner.propTypes = {
    imagePath: PropTypes.string,
    alt: PropTypes.string,
    style: PropTypes.object,
    classNameWrapper: PropTypes.string,
    classNameImg: PropTypes.string
};

Spinner.defaultProps = {
    imagePath: '/images/loader.gif',
    alt: 'Loading...',
    style: { width: '80px', margin: '60px', display: 'block' }
};

export default Spinner;