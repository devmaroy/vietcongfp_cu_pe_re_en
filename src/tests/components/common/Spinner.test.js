import React from 'react';
import { shallow } from 'enzyme';
import Spinner from '../../../components/common/Spinner';

test( 'should correctly render Spinner', () => {
    const wrapper = shallow( <Spinner /> );

    expect( wrapper ).toMatchSnapshot();
});