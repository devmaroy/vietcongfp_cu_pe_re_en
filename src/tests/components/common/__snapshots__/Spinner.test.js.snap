// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Spinner 1`] = `
<div>
  <img
    alt="Loading..."
    src="/images/loader.gif"
    style={
      Object {
        "display": "block",
        "margin": "60px",
        "width": "80px",
      }
    }
  />
</div>
`;
